src/algorithms/tonal/pitchyinprobabilistic.h

```diff
@@ -0,0 +1,133 @@
+
+#include "algorithmfactory.h"
+#include "network.h"
+#include "pool.h"
+#include "streamingalgorithmcomposite.h"
+
+namespace essentia {
+namespace streaming {
+
+class PitchYinProbabilistic : public AlgorithmComposite {
+
+ protected:
+  // Algorithm* _loudnessEBUR128Filter;
+  Algorithm* _frameCutter;
+  Algorithm* _yinProbabilities;
+  standard::Algorithm* _yinProbabilitiesHMM;
+
+  SinkProxy<Real> _signal;
+  Source<std::vector<Real> > _pitch;
+  Source<std::vector<Real> > _voicedProbs;
+
+  Pool _pool;
+
+  int _frameSize;
+  int _hopSize;
+  Real _lowAmp;
+  int _outputUnvoiced;
+  bool _precisetime;
+
+  scheduler::Network* _network;
+
+
+ public:
+  PitchYinProbabilistic();
+   ~PitchYinProbabilistic();
+
+  void declareProcessOrder() {
+    declareProcessStep(ChainFrom(_frameCutter));
+    declareProcessStep(SingleShot(this));
+  }
+
+  void declareParameters() {
+    declareParameter("sampleRate", "the sampling rate of the audio signal [Hz]", "(0,inf)", 44100.);
+    declareParameter("frameSize", "the frame size of FFT", "(0, inf)", 2048);
+    declareParameter("hopSize", "the hop size with which the pitch is computed", "[1,inf)", 256); 
+    declareParameter("lowAmp", "the low RMS amplitude threshold", "(0,1]", 0.1);  
+    declareParameter("outputUnvoiced", "whether output unvoiced frame. 0: output non-voiced pitch as 0.; 1: output non-voiced pitch as absolute values; 2: output non-voiced pitch as negative values", "{0,1,2}", 2);
+    declareParameter("precisetime", "use non-standard precise YIN timing (slow).", "{true,false}", false);
+  };
+
+  void configure();
+  AlgorithmStatus process();
+  void reset();
+
+  static const char* name;
+  static const char* category;
+  static const char* description;
+};
+
+} // namespace streaming
+} // namespace essentia
+
+
+#include "vectorinput.h"
+
+namespace essentia {
+namespace standard {
+
+class PitchYinProbabilistic : public Algorithm {
+ protected:
+  Input<std::vector<Real> > _signal;
+  Output<std::vector<Real> > _pitch;
+  Output<std::vector<Real> > _voicedProbs;
+
+  streaming::Algorithm* _PitchYinProbabilistic;
+  streaming::VectorInput<Real>* _vectorInput;
+  scheduler::Network* _network;
+  Pool _pool;
+
+  int _frameSize;
+  int _hopSize;
+  Real _lowAmp;
+  int _outputUnvoiced;
+
+ public:
+
+  PitchYinProbabilistic();
+  ~PitchYinProbabilistic();
+
+  void declareParameters() {
+    declareParameter("sampleRate", "the sampling rate of the audio signal [Hz]", "(0,inf)", 44100.);
+    declareParameter("frameSize", "the frame size of FFT", "(0, inf)", 2048);
+    declareParameter("hopSize", "the hop size with which the loudness is computed", "[1,inf)", 256);
+    declareParameter("lowAmp", "the low RMS amplitude threshold", "(0,1]", 0.1);  
+    declareParameter("outputUnvoiced", "whether output unvoiced frame, 0: output non-voiced pitch as 0.; 1: output non-voiced pitch as absolute values; 2: output non-voiced pitch as negative values", "{0,1,2}", 2);
```

Comment:
Let's use string instead for more clarity: "zero", "abs", "negative"

Reply:
It is in fact more clear to me outputUnvoiced here. zero, abs, negative are the values how we output unvoiced frames.

Reply:
Correct, I refer to that we should use strings instead of numbers as the parameter values (see the "method" parameter [here](https://essentia.upf.edu/documentation/reference/streaming_OnsetDetection.html) as an example)